doc/protocol.md

```diff
@@ -1025,6 +1026,38 @@ message.
 Param: `<timestamp>` is a copy of the timestamp carried by the corresponding
 [PING](#ping) message.
 
+### LOG_MESSAGE
+
+    head = 'log_message' <session ident> <message size>
+    body = <message>
+
+The log message is a debug message sent by the server that can be received asynchronously at
+any time after the websocket ACCEPT response and prints a message from the server in the
+client debug logs. The level value in the message JSON data indicates the "severity" of the
+message and translates to the debug log level for which this message will be printed in the
+client debug log.
+
+Param: `<session ident>` is an integer value that specifies which session should print this
+log message. This value can be `0` if the log message applies to the connection.
+
+The body of the message is required and will be in JSON format with the following keys
+currently supported:
+    - `co_id` is a string value that provides the appservices request/connection id for the
```

Comment:
I wonder if this should be part of the header cc @kmorkos

Reply:
hmm, i think if the client is considering it optional then we should keep it in the json blob

Reply:
I think it's up to the server if it's optional or not.